service/src/main/java/org/cbioportal/genome_nexus/service/internal/MutationAssessorServiceImpl.java

```diff
@@ -71,26 +77,27 @@ public MutationAssessor getMutationAssessor(VariantAnnotation annotation)
             throw new MutationAssessorNotFoundException(annotation.getVariant());
         }
 
-        return this.getMutationAssessor(buildRequest(annotation), annotation.getVariant());
+        MutationAssessor mutationAssessor = this.getMutationAssessorByMutationAssessorVariant(buildRequest(annotation));
 
+        // add original hgvs variant value too
+        mutationAssessor.setHgvs(annotation.getVariant());
+
+        return mutationAssessor;
     }
 
-    public MutationAssessor getMutationAssessor(String variant, String hgvs)
+    /**
+     * @param variant   mutation assessor variant (ex: 7,140453136,A,T)
+     */
+    public MutationAssessor getMutationAssessorByMutationAssessorVariant(String variant)
         throws MutationAssessorNotFoundException, MutationAssessorWebServiceException
     {
-        MutationAssessor mutationAssessorObj = null;
+        MutationAssessor mutationAssessor = null;
 
         try {
-            List<MutationAssessor> list = this.externalResourceFetcher.fetchInstances(variant);
-
-            if (list.size() != 0) {
-                mutationAssessorObj = list.get(0);
-            }
+            // get the annotation from the web service and save it to the DB
+            mutationAssessor = cachedExternalResourceFetcher.fetchAndCache(variant);
 
-            if (mutationAssessorObj != null) {
-                mutationAssessorObj.setInput(hgvs);
```

Comment:
This is where we were overwriting the actual input value with hgvs notation. This is not a good practice :)